ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/util/DebugDataDumper.java

```diff
@@ -0,0 +1,157 @@
+        "gossip message with decoding error", topicPath.resolve(fileName), originalMessage);
+  }
+
+  public void saveGossipRejectedMessageToFile(
+      final String topic, final Optional<UInt64> arrivalTimestamp, final Bytes decodedMessage) {
+    if (!enabled) {
+      return;
+    }
+    final String fileName = String.format("%s.ssz", formatTimestamp(arrivalTimestamp));
+    final Path topicPath = Path.of(GOSSIP_MESSAGES_DIR).resolve(REJECTED_SUB_DIR).resolve(topic);
+    saveBytesToFile("rejected gossip message", topicPath.resolve(fileName), decodedMessage);
+  }
+
+  public void saveInvalidBlockToFile(
+      final UInt64 slot, final Bytes32 blockRoot, final Bytes blockSsz) {
+    if (!enabled) {
+      return;
+    }
+    final String fileName = String.format("%s_%s.ssz", slot, blockRoot.toUnprefixedHexString());
+    saveBytesToFile("invalid block", Path.of(INVALID_BLOCK_DIR).resolve(fileName), blockSsz);
+  }
+
+  @VisibleForTesting
+  protected void saveBytesToFile(
+      final String object, final Path relativeFilePath, final Bytes bytes) {
+    final Path path = directory.resolve(relativeFilePath);
+    try {
+      Files.write(path, bytes.toArray());
+      LOG.info("Saved {} bytes to file. Location: {}", object, relativeFilePath);
+    } catch (NoSuchFileException e) {
+      if (!path.getParent().toFile().mkdirs()) {
+        LOG.error("Failed to save {} bytes to file.", object, e);
+        return;
+      }
+      saveAfterCreatingTopicDirectory(object, relativeFilePath, bytes);
+    } catch (IOException e) {
+      LOG.error("Failed to save {} bytes to file.", object, e);
+    }
+  }
+
+  private void saveAfterCreatingTopicDirectory(
+      final String object, final Path relativeFilePath, final Bytes bytes) {
+    final Path path = directory.resolve(relativeFilePath);
+    try {
+      Files.write(path, bytes.toArray());
+      LOG.info("Saved {} bytes to file. Location: {}", object, relativeFilePath);
+    } catch (IOException e) {
+      LOG.error("Failed to save {} bytes to file.", object, e);
+      if (!path.getParent().toFile().exists()) {
+        LOG.error(
+            "{} directory does not exist. Disabling saving debug data to file.",
+            relativeFilePath.getParent());
+      }
+    }
+  }
+
+  private void createDirectory(final Path path, final String directoryName, final String object) {
+    if (!enabled) {
+      return;
+    }
+    if (path.toFile().mkdirs()) {
+      LOG.debug("{} directory has been created to save {}.", directoryName, object);
+    } else {
+      if (!path.toFile().exists()) {
+        this.enabled = false;
+        LOG.error(
+            "Unable to create {} directory to save {}. Disabling saving debug data to file.",
+            directoryName,
+            object);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  protected static String formatTimestamp(final Optional<UInt64> arrivalTimestamp) {
+    if (arrivalTimestamp.isEmpty()) {
+      return "unknown";
+    }
```

Comment:
why static also?

Reply:
So I don't need to create a manager to use it in the tests

Reply:
Can change it if needed?